src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy

```diff
@@ -15,16 +15,18 @@ class ConsulProcess implements AutoCloseable {
     private final ConsulPorts consulPorts
     private final String address
     private final Process process
+    private final ConsulLogHandler consulLogHandler
     private final SimpleConsulClient simpleConsulClient
 
-    ConsulProcess(Path dataDir, ConsulPorts consulPorts, String address, Process process) {
+    ConsulProcess(Path dataDir, ConsulPorts consulPorts, String address, Process process, ConsulLogHandler consulLogHandler) {
         this.dataDir = dataDir
         this.consulPorts = consulPorts
         this.address = address
         this.process = process
+        this.consulLogHandler = consulLogHandler
         this.simpleConsulClient = new SimpleConsulClient(address, httpPort)
 
-        addShutdownHook { this.process.destroyForcibly()}
+        addShutdownHook { this.process.destroyForcibly() }
```

Comment:
@pszymczyk in the first commit you had consulLogHandler.close() here, might me instead we should invoke close() method to close all resources?

Reply:
Yes first I close the `consulLogHandler` here also, but this shutdown hook is added in case of intermediate vm shutdown we want to close other process (not managed by jvm). `ConsulLogHandler` is simple thread pool so it's managed b vm so when vm goes down this thread pool will also go down. Do I misunderstood something? please correct me if I'm wrong :)

Reply:
Sounds about right. The main goal of shutdown hook is to close external resources.